pkg/vm/vm_test.go

```diff
@@ -960,6 +971,198 @@ func TestAssertEqualInstruction(t *testing.T) {
+		setInitialReg(vm, 1, 1 ,0)
+
+		err:= vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Ap, uint64(124))
+	})
+}
+
+func TestJumpInstruction(t *testing.T) {
+	hintrunner := noHintRunner{}
+	setInitialReg := func(vm *VirtualMachine, regvals ...uint64) {
+		if len(regvals) != 3 {
+			panic("expected three register values")
+		}
+		vm.Context.Ap = regvals[0]
+		vm.Context.Fp = regvals[1]
+		vm.Context.Pc = mem.MemoryAddress{SegmentIndex: 0, Offset: regvals[2]}
+	}
+
+	t.Run("test abs jump with immediate", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp abs 15;")
+		setInitialReg(vm, 1, 1, 0)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc.Offset, uint64(15))
+	})
+	t.Run("test abs jump with address", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp abs [ap];")
+		setInitialReg(vm, 1, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Ap, &mem.MemoryAddress{SegmentIndex: 15, Offset: 18})
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc, mem.MemoryAddress{SegmentIndex: 15, Offset: 18})
+	})
+
+	t.Run("test rel jump with address", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp rel [ap + 1] + [fp];")
+		setInitialReg(vm, 1, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Ap+1, 1)
+		writeToDataSegment(vm, vm.Context.Fp, 7)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc.Offset, uint64(8))
+
+	})
+
+	t.Run("test rel jump with immediate", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp rel 123;")
+		setInitialReg(vm, 1, 1, 0)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc.Offset, uint64(123))
+
+	})
+
+	t.Run("test conditional jump, if <op> != 0", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp rel [ap - 1] if [fp + 2] != 0;")
+		setInitialReg(vm, 1, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Ap-1, 4)
+		writeToDataSegment(vm, vm.Context.Fp+2, 5)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc.Offset, uint64(4))
+
+	})
+
+	t.Run("test conditional jump, if <op> == 0", func(t *testing.T) {
+		vm := defaultVirtualMachineWithCode("jmp rel [ap - 1] if [fp + 2] != 0;")
+		setInitialReg(vm, 1, 1, 0)
+
+		writeToDataSegment(vm, vm.Context.Ap-1, 3)
+		writeToDataSegment(vm, vm.Context.Fp+2, 0)
+
+		err := vm.RunStep(&hintrunner)
+		require.NoError(t, err)
+
+		assert.Equal(t, vm.Context.Pc.Offset, uint64(1))
+
+	})
+
+}
+
+func TestCallAndRetInstructions(t *testing.T){
+	hintrunner := noHintRunner{}
+	setInitialReg := func(vm *VirtualMachine, regvals ...uint64) {
+		if len(regvals) != 3 {
+			panic("expected three register values")
+		}
+		vm.Context.Ap = regvals[0]
+		vm.Context.Fp = regvals[1]
+		vm.Context.Pc = mem.MemoryAddress{SegmentIndex: 0, Offset: regvals[2]}
+	}
```

Comment:
You have the same function repeated across all tests. I think is good to only define it once

Reply:
Noted